File: two pointers/merge_sorted_array.java
import java.util.Arrays;


public class merge_sorted_array {
    //brute force
    // tc O(nlogn), sc O(n)
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        for(int i=0;i<n;i++){
            nums1[m+i] = nums2[i];
        }
        Arrays.sort(nums1);
    }

    //two pointers
    // tc O(n), sc O(n)
    public void merge2(int[] nums1, int m, int[] nums2, int n) {
        int l=m-1, r=m+n-1, i=n-1;
        while(i>=0){
            if(l>=0 && nums1[l]>nums2[i]){
                nums1[r] = nums1[l--];
            }
            else{
                nums1[r] = nums2[i--];
            }
            r--;
        }

    }
}

File: two pointers/merge_sorted_array.py
# brute force
# tc O(nlogn), sc O(1)
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """
        for i in range(n):
            nums1[m+i] = nums2[i]
        nums1.sort()

# two pointers
# tc O(n), sc O(1)
class Solution2:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        l, r, i = m-1, m+n-1, n-1
        while i>=0:
            if l>=0 and nums1[l]>nums2[i]:
                nums1[r] = nums1[l]
                l -= 1
            else:
                nums1[r] = nums2[i]
                i -= 1
            r -= 1
        